### PR TITLE
[WIP] feat: ChugSplash v0

### PR DESCRIPTION
**Pull request by smartcontracts**
**NOT READY FOR REVIEW**

(I'm just using this as a way to keep track of my diff)

**Description**
Working on the various things necessary to get chugsplash off the ground.

### CODE REVIEW

**Comment by gakonst** [comment]
This PR does a lot of things, if you have specific things that are not chugsplash related, please split them in a separate PR!

**Comment by gakonst** [comment]
Closing this until there's further activity!

File: packages/contracts/contracts/chugsplash/ChugSplashDeployer.sol [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        currentBundleHash = _bundleHash;
+        currentBundleSize = _bundleSize;
+        currentBundleTxsExecuted = 0;
+    }
+
+    function executeAction(
+        ActionType _type,
+        string memory _target,
+        bytes memory _data,
+        uint256 _gasLimit
```
Comment (karlfloersch) [inline packages/contracts/contracts/chugsplash/ChugSplashDeployer.sol:89]: Consider adding a `cancelDeployment` function & requiring all calls pass

File: packages/contracts/contracts/chugsplash/ChugSplashDeployer.sol [hunk 2]
```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: MIT
+// @unsupported: ovm
+pragma solidity >0.5.0 <0.8.0;
+
+import { ChugSplashProxy } from "./ChugSplashProxy.sol";
+
+/**
+ * @title ChugSplashDeployer
+ */
+contract ChugSplashDeployer {
```
Comment (karlfloersch) [inline packages/contracts/contracts/chugsplash/ChugSplashDeployer.sol:10]: Consider tradeoff of shutting down the system being encoded into the contract vs in the bundle

**Pull request closed without merging**